aaq/utils.py

```diff
@@ -35,8 +35,8 @@ def search(query_text, generate_llm_response, query_metadata):
 
     response = requests.request("POST", url, json=payload, headers=headers)
 
-    if "error" in response.json():
-        error_detail = response.json()["error"].get("detail", "")
+    if "detail" in response.json():
```

Comment:
Are we sure detail won't be present for any other requests?

Reply:
a successful search response doesn't doesn't have the detail key

Reply:
Cool, but it might be confusing later on so what about something like:
`if response.status_code == 400 "detail" in response.json():`

Then theres no doubt that we are handling an error,

You'll need to make sure you mock the status code the same as the AAQ service returns it.